cast/src/lib.rs

```diff
@@ -308,6 +308,46 @@ where
     ) -> Result<String> {
         Ok(format!("{}", self.provider.get_code(who, block).await?))
     }
+
+    /// ```no_run
+    /// use cast::Cast;
+    /// use ethers_providers::{Provider, Http};
+    /// use std::convert::TryFrom;
+    ///
+    /// # async fn foo() -> eyre::Result<()> {
+    /// let provider = Provider::<Http>::try_from("http://localhost:8545")?;
+    /// let cast = Cast::new(provider);
+    /// let tx_hash = "0xf8d1713ea15a81482958fb7ddf884baee8d3bcc478c5f2f604e008dc788ee4fc";
+    /// let tx = cast.transaction(tx_hash.to_string(), None, false).await?;
+    /// println!("{}", tx);
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub async fn transaction(
+        &self,
+        tx_hash: String,
+        field: Option<String>,
+        to_json: bool,
+    ) -> Result<String> {
+        let transaction_result = self
+            .provider
+            .get_transaction(H256::from_str(&tx_hash)?)
+            .await?
+            .ok_or_else(|| eyre::eyre!("transaction {:?} not found", tx_hash))?;
+
+        let transaction = if let Some(ref field) = field {
+            serde_json::to_value(&transaction_result)?
+                .get(field)
+                .cloned()
```

Comment:
there should be `Value::take(field)` which removes the value, so we could get rid of the clone here

Reply:
eh in this case i guess the clone is not that expensive, one-time call expected anyway